controllers/routemap_test.go

```diff
@@ -214,3 +218,214 @@ func TestGetObservedGeneration(t *testing.T) {
 		assert.Equal(t, tt.ok, o)
 	}
 }
+
+type testInformer struct {
```

Comment:
We have a standard way to mock the Kubernetes client, populate it with objects, and call its APIs: please see [here](https://github.com/iter8-tools/iter8/blob/master/controllers/k8sclient/fake/simple_test.go).

Could we please use the fake client instead of creating our own Kubernetes mocking mechanisms?

Reply:
A more end-to-end example of how fake client is used is here: https://github.com/iter8-tools/iter8/blob/master/controllers/allcontrollers_test.go